File: client/src/features/suspect/types/suspect.ts
export enum SuspectStatus {
    Unknown = 0,
    Detained = 1,
    Fugitive = 2,
    Killed = 3,
    Sentenced = 4,
    Released = 5,
}

export const SuspectStatusLabels: Record<SuspectStatus, string> = {
    [SuspectStatus.Unknown]: 'غير معروف',
    [SuspectStatus.Detained]: 'موقوف',
    [SuspectStatus.Fugitive]: 'هارب',
    [SuspectStatus.Killed]: 'مقتول',
    [SuspectStatus.Sentenced]: 'محكوم',
    [SuspectStatus.Released]: 'مُفرج عنه',
};

export enum MaritalStatus {
    Single = 1,
    Married = 2,
    Divorced = 3,
    Widower = 4,
}

export const MaritalStatusLabels: Record<MaritalStatus, string> = {
    [MaritalStatus.Single]: 'أعزب',
    [MaritalStatus.Married]: 'متزوج',
    [MaritalStatus.Divorced]: 'مطلق',
    [MaritalStatus.Widower]: 'أرمل',
};

export interface UpdateSuspectRequest {
    suspectId?: string;
    firstName?: string;
    secondName?: string;
    thirdName?: string;
    fourthName?: string;
    fivthName?: string;
    kunya?: string;
    codeNum?: string;
    motherName?: string;
    dateOfBirth?: string;
    placeOfBirth?: string;
    tribe?: string;
    maritalStatus?: MaritalStatus;
    wivesCount?: number;
    childrenCount?: number;
    legalArticle?: string;
    healthStatus?: string;
    photoUrl?: string;
    status?: SuspectStatus;
}

export interface UpdateSuspectStatusRequest {
    suspectId: string;
    newStatus: SuspectStatus;
    notes: any;
}

export interface Suspect {
    addresses: Address[];
    bayahDetails: BayahDetail[];
    caseInvolvements: CaseInvolvement[];
    childrenCount: number;
    contacts: Contact[];
    createdAt: string;
    currentStatus: SuspectStatus;
    currentStatusName: string;
    dateOfBirth?: string;
    familyName: string;
    firstName: string;
    fivthName?: string;
    fourthName?: string;
    fullName: string;
    healthStatus?: any;
    id: string;
    kunya: string;
    legalArticle: string;
    lifeHistories: LifeHistory[];
    maritalStatus?: any;
    maritalStatusName?: any;
    motherName: string;
    operations: Operation[];
    organizationalAssignments: OrganizationalAssignment[];
    photoUrl: string | null;
    placeOfBirth: string;
    relativesAndAssociates: RelativesAndAssociate[];
    secondName: string;
    thirdName: string;
    trainingCourses: TrainingCourse[];
    tribe?: string;
    wivesCount: number;
}


export interface CreateSuspectRequest {
   
  firstName?: string;
  secondName?: string;
  thirdName?: string;
  fourthName?: string;
  fivthName?: string;
  codeNum?: string;
  kunya?: string;
  motherName?: string;
  dateOfBirth?: string;
  placeOfBirth?: string;
  tribe?: string;
  status?: SuspectStatus;
  legalArticle?: string;

}



export interface SuspectQuery {
    page?: number;
    pageSize?: number;
    searchTerm?: string;
}

export interface NetworkSuspect {
    nodes: NetworkSuspectNode[]
    links: NetworkSuspectLink[]
}

export interface NetworkSuspectNode {
    id: string
    label: string
    type: string
    group: string
}

export interface NetworkSuspectLink {
    source: string
    target: string
    label: string
    type: string
}


// Suspect Details

export interface SuspectDetails {
    id: string
    firstName: string
    secondName: string
    thirdName: string
    fourthName: any
    fivthName: string
    familyName: any
    fullName: string
    codeNum: string
    kunya: any
    motherName: any
    dateOfBirth: any
    placeOfBirth: any
    tribe: any
    maritalStatus: any
    maritalStatusName: any
    wivesCount: number
    childrenCount: number
    legalArticle: any
    healthStatus: any
    photoUrl: any
    currentStatus: number
    currentStatusName: any
    createdAt: string
    addresses: Address[]
    contacts: Contact[]
    relativesAndAssociates: RelativesAndAssociate[]
    lifeHistories: LifeHistory[]
    bayahDetails: BayahDetail[]
    trainingCourses: TrainingCourse[]
    operations: Operation[]
    organizationalAssignments: OrganizationalAssignment[]
    caseInvolvements: CaseInvolvement[]
}

export interface Address {
    id: string
    type: number
    typeName: string
    city: string
    district: string
    details: any
    gpsCoordinates: any
}

export interface Contact {
    id: string
    type: number
    typeName: string
    value: string
    ownerName: any
}

export interface RelativesAndAssociate {
    id: string
    fullName: string
    relationship: number
    relationshipName: string
    spouseName: any
    currentStatus: any
    notes: any
}

export interface LifeHistory {
    id: string
    educationLevel: any
    schoolsAttended: any
    civilianJobs: any
    radicalizationStory: any
}

export interface BayahDetail {
    id: string
    date: any
    location: any
    recruiterName: any
    textOfPledge: any
}

export interface TrainingCourse {
    id: string
    courseType: number
    courseTypeName: string
    location: any
    trainerName: any
    classmates: any
}

export interface Operation {
    id: string
    operationType: number
    operationTypeName: string
    date: any
    location: any
    roleInOp: any
    associates: any
}

export interface OrganizationalAssignment {
    id: string
    orgUnitId: string
    orgUnitName: string
    directCommanderId: any
    directCommanderName: any
    roleTitle: number
    roleTitleName: string
    startDate: string
    endDate: any
}

export interface CaseInvolvement {
    id: string
    caseId: string
    caseFileNumber: string
    caseTitle: string
    accusationType: number
    accusationTypeName: string
    legalStatus: number
    legalStatusName: string
    confessionDate: any
    notes: any
}

export interface LifeHistoryRequest {
    suspectId?: string
    educationLevel?: string
    schoolsAttended?: string
    civilianJobs?: string
    radicalizationStory?: string
    lifeHistoryId?: string
}

File: NEXUS/Data/Entities/Suspect.cs
using NEXUS.Data.Common;
using NEXUS.Data.Enums;
using System.ComponentModel.DataAnnotations;

namespace NEXUS.Data.Entities;

public class Suspect : BaseEntity
{
    [MaxLength(25)]
    public string FirstName { get; set; } = string.Empty;
    [MaxLength(25)]
    public string SecondName { get; set; } = string.Empty;
    [MaxLength(25)]
    public string ThirdName { get; set; } = string.Empty;
    [MaxLength(25)]
    public string? FourthName { get; set; }
    [MaxLength(25)]
    public string? FivthName { get; set; } = string.Empty;
    [Required, MaxLength(255)]
    public string FullName { get; set; } = string.Empty;  // الاسم الكامل

    [MaxLength(100)]
    public string? CodeNum { get; set; }

    [MaxLength(100)]
    public string? Kunya { get; set; } // الكنية (أبو فلان)

    [MaxLength(100)]
    public string? MotherName { get; set; }

    public DateOnly? DateOfBirth { get; set; }
    [MaxLength(255)]
    public string? PlaceOfBirth { get; set; }
    [MaxLength(40)]
    public string? Tribe { get; set; } // العشيرة
    public MaritalStatus? MaritalStatus { get; set; }
    public int WivesCount { get; set; }
    public int ChildrenCount { get; set; }

    [MaxLength(100)]
    public string? LegalArticle { get; set; } // المادة القانونية (4 إرهاب)
    [MaxLength(100)]
    public string? HealthStatus { get; set; }
    [MaxLength(250)]
    public string? PhotoUrl { get; set; }

    [Required]
    public SuspectStatus CurrentStatus { get; set; } // (موقوف، هارب، مقتول، محكوم)

    // --- Navigation Properties (العلاقات) ---
    public ICollection<Address> Addresses { get; set; } = default!;
    public ICollection<Contact> Contacts { get; set; } = default!;
    public ICollection<RelatedPerson> RelativesAndAssociates { get; set; } = default!;
    public ICollection<LifeHistory> LifeHistories { get; set; } = default!;

    // العلاقات التنظيمية
    public ICollection<BayahDetail> BayahDetails { get; set; } = default!;
    public ICollection<TrainingCourse> TrainingCourses { get; set; } = default!;
    public ICollection<Operation> Operations { get; set; } = default!;
    public ICollection<Assignment> OrganizationalAssignments { get; set; } = default!; // المناصب التي شغلها

    // القضايا
    public ICollection<CaseSuspect> CaseInvolvements { get; set; } = default!;

    public void UpdateFullName()
    {
        FullName = string.Join(" ", new[] { FirstName, SecondName, ThirdName, FourthName, FivthName, Tribe }
            .Where(s => !string.IsNullOrWhiteSpace(s))).Trim();
    }
}
